lib/publiccloud/utils.pm

```diff
@@ -192,6 +194,10 @@ sub is_hardened() {
     return is_public_cloud && get_var('FLAVOR') =~ 'Hardened';
 }
 
+sub cloudinit_enabled {
+    return (is_public_cloud && (is_azure || is_ec2) && !is_sle_micro);
```

Comment:
```suggestion
    return ((is_azure || is_ec2) && !is_sle_micro);
```

if you open `is_ec2` and `is_azure` implementation you will see that calling `is_publiccloud` is not needed

Reply:
Please re-check now.